scripts/idl/matter_grammar.lark

```diff
@@ -63,23 +71,21 @@ struct_field: member_attribute* field
 member_attribute: "optional"i -> optional
                 | "nullable"i -> nullable
 
-field: data_type id list_marker? "=" number ";"
+field: data_type id list_marker? "=" number
```

Comment:
You should have a different rule for an integer, versus a number. Default values can be a `number` (real or integer) but many other pieces can only be integers

Reply:
Currently number == integer in the IDL (`number: POSITIVE_INTEGER | HEX_INTEGER`). I can do renaming of number to integer in a followup PR rather than this one.